### DISCUSSION TITLE
Permission Scaling API

### DISCUSSION THREAD

**Discussion by ItsSynchro** [QuickShop-Community/QuickShop-Hikari#995] [Ideas]
Hi,

I'm looking to create a coin shop for players to be able to purchase bonus chestshops, however I've ran into the issue where due to there being no api for what I'm looking to do, its pretty impossible. 

If someone were to start at Merchant rank, which has 15 chestshops available, then upgrade to the next rank, the highest permission nodes would just take priority and pretty much reset all progress made with the coin shop. On top of that there's the confusion of setting permissions normally in the plugin for the ranks themselves.

**Comment by Ghost-chu** [comment]
Hi!
Unfortunately, I have been unable to work on the plugin for the last 15-20 days due to health issues. When I recover I will check and work on the backlog.

If you mean user upgrade, you can always give players a new permission that have a higher limit of shops amount, quickshop will always pick the highest one.
See: https://ghost-chu.github.io/QuickShop-Hikari-Documents/docs/modules/shops/ranks

Feel free to correct me if I've misunderstood this thread and I'll check it out when I get back :)

**Comment by Ghost-chu** [comment]
Are you looking for the ability to dynamically add quota to players?  For example, instead of using a preset fixed value, the quota will be increased dynamically for players on the basis of the original in multiple times.

**Comment by ItsSynchro** [comment]
An example might clarify my idea a bit.

Currently on my server we allow players to unlock a set amount of chest shops at certain ranks. However we'd love to give them the ability to unlock the ability to place additional shops through things like in game cash or quests.
Currently the only way we have to do this is to set each additional chest as its own permission and add a higher permission to someone everytime they unlock more shops.

However, the big downside with this is that if they've unlocked these additional shops through in game cash, then afterwards buy a rank that comes with a higher amount of chest shops, those additional shops will not be available to the player anymore.
